base/special/trig.jl

```diff
@@ -76,6 +76,105 @@ function cos_kernel(x::DoubleFloat32)
     Float32(((1.0+z*C0) + w*C1) + (w*z)*r)
 end
 
+# asin methods
+ASIN_X_MIN_THRESHOLD(::Type{Float32}) = 2.0f0^-12
+ASIN_X_MIN_THRESHOLD(::Type{Float64}) = sqrt(eps(Float64))
+
+asin_p(t::Float64) = t*@horner(t,
+                             1.66666666666666657415e-01,
+                            -3.25565818622400915405e-01,
+                             2.01212532134862925881e-01,
+                            -4.00555345006794114027e-02,
+                             7.91534994289814532176e-04,
+                             3.47933107596021167570e-05)
+
+asin_q(t::Float64) = @horner(t,
+                             1.0,
+                            -2.40339491173441421878e+00,
+                             2.02094576023350569471e+00,
+                            -6.88283971605453293030e-01,
+                             7.70381505559019352791e-02)
+
+asin_p(t::Float32) = t*@horner(t,
+                             1.6666586697f-01,
+                             -4.2743422091f-02,
+                             -8.6563630030f-03)
+
+asin_q(t::Float32) = muladd(t, -7.0662963390f-01, 1.0f0)
+
+@inline function asin_kernel(t::Float64, x)
+    pio2_lo = 6.12323399573676603587e-17
+    s = sqrt(t)
```

Comment:
Don't know, are you just curious, or is there something like a `pos_sqrt(t)` function?